vnc_lite.html

```diff
@@ -22,156 +20,171 @@
+        }
+
+        #noVNC_buttons {
+            padding: 1px;
+            flex: 1;
+            display: flex;
+            justify-content: flex-end;
+        }
 
-    <!-- Stylesheets -->
-    <link rel="stylesheet" href="app/styles/lite.css">
+    </style>
 
-    <!-- promise polyfills promises for IE11 -->
+    <!-- Promise polyfill for IE11 -->
     <script src="vendor/promise.js"></script>
+
     <!-- ES2015/ES6 modules polyfill -->
     <script type="module">
         window._noVNC_has_module_support = true;
     </script>
     <script>
         window.addEventListener("load", function() {
             if (window._noVNC_has_module_support) return;
-            var loader = document.createElement("script");
+            const loader = document.createElement("script");
             loader.src = "vendor/browser-es-module-loader/dist/browser-es-module-loader.js";
             document.head.appendChild(loader);
         });
     </script>
 
     <!-- actual script modules -->
     <script type="module" crossorigin="anonymous">
-        // Load supporting scripts
-        import * as WebUtil from './app/webutil.js';
+        // RFB holds the API to connect and communicate with a VNC server
         import RFB from './core/rfb.js';
 
-        var rfb;
-        var desktopName;
+        let rfb;
+        let desktopName;
 
-        function updateDesktopName(e) {
-            desktopName = e.detail.name;
+        // When this function is called we have
+        // successfully connected to a server
+        function connectedToServer(e) {
+            document.getElementById('sendCtrlAltDelButton').disabled = false;
+            status("Connected to " + desktopName, "normal");
+        }
+
+        // This function is called when we are disconnected
+        function disconnectedFromServer(e) {
+            document.getElementById('sendCtrlAltDelButton').disabled = true;
+            if (e.detail.clean) {
+                status("Disconnected", "normal");
+            } else {
+                status("Something went wrong, connection is closed", "error");
+            }
         }
-        function credentials(e) {
-            var html;
 
-            var form = document.createElement('form');
+        // When this function is called, the server requires
+        // credentials to authenticate
+        function credentialsAreRequired(e) {
+            status('', 'warn');
+
+            // Let's create a password input
+            const form = document.createElement('form');
             form.innerHTML = '<label></label>';
             form.innerHTML += '<input type=password size=10 id="password_input">';
             form.onsubmit = setPassword;
+            // And add place it in the status-bar
+            const statusElem = document.getElementById('noVNC_status');
+            statusElem.innerHTML = '';
+            statusElem.appendChild(form);
+            statusElem.querySelector('label').textContent = 'Password Required: ';
```

Comment:
Can't we use `window.prompt()`? It would make things simpler.

Reply:
I had a look at that as well, but it doesn't allow us to hide the input as the user is typing it, as is normal for password inputs

Reply:
Hmmm... that's not ideal. But perhaps it's still worth it to get rid of a bunch of complexity? It is just an example after all.

Reply:
Yeah, I think so too